docs: add commands (#105)
canonical/rockcraft: docs/reference.rst

NOTE: at the moment, it is not possible to mix packages and slices in the same stage-packages field.

Rockcraft commands
------------------
Lifecycle commands
..................
Lifecycle commands can take an optional parameter ``<part-name>``. When a part name is provided, the command applies to the specific part. When no part name is provided, the command applies to all parts.

clean
^^^^^
Removes a part's assets. When no part is provided, the entire build environment (e.g. the LXD instance) is removed

pull
^^^^

> Downloads or retrieves artefacts defined for each part.

Reviewer: ```suggestion
Downloads or retrieves artifacts defined for each part.
```

overlay
^^^^^^^

> Creates part layers over the base filesystem.

Reviewer: ```suggestion
Execute operations defined for each part on a layer over the base filesystem, potentially modifying its contents.
```

build
^^^^^

> Builds artefacts defined for each part.

Reviewer: ```suggestion
Builds artifacts defined for each part.
```

stage
^^^^^

> Stages built artefacts into a common staging area.

Reviewer: ```suggestion
Stages built artifacts into a common staging area, for each part.
```

prime
^^^^^

> Primes artefacts defined for each part.

Reviewer: ```suggestion
Prepare, for each part, the final payload to be packed as a ROCK, performing additional processing and adding metadata files.
```

pack
^^^^

> Creates a ROCK from the primed artefacts.

Reviewer: ```suggestion
*This is the default lifecycle command for ``rockcraft``.*

Process parts and create the ROCK as an OCI archive file containing the project payload with the provided metadata.
```

Other commands
..............
init
^^^^

> Initializes a rockcraft project with a boilerplate ``rockcraft.yaml`` configuration file.

Reviewer: ```suggestion
Initializes a rockcraft project with a boilerplate ``rockcraft.yaml`` file.
```

NOTE: at the moment, it is not possible to mix packages and slices in the same stage-packages field.

Rockcraft commands
------------------

> Lifecycle commands

Reviewer 2: can you screenshot what this new section looks like, rendered? (You might need to update your doc env, there's a new theme)

Author: Sure thing!

Reviewer 2: Thanks! Is there a particular reason why you chose bulleted lists? I think the docs might flow better if we use regular paragraphs for each section, even if they're just a single sentence.
I understand that this is an initial pass to have "something" there, so thanks for doing this regardless ;)

Author: No particular reason.  I agree that paragraphs would be better.  I updated the docs and the rendered screenshot.
